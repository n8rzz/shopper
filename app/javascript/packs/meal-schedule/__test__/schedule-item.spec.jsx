import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import { ScheduleItem } from '../schedule-item.component';
import {
    assemblyMock,
    itemMock,
    mealScheduleItemMock,
} from '../../__mocks__/mocks';

describe('ScheduleItem', () => {
    describe('when passed an assembly', () => {
        test('renders correctly with valid props', () => {
            const component = shallow(<ScheduleItem
                mealScheduleId={mealScheduleItemMock.id}
                isAssembly={true}
                scheduleItem={assemblyMock}
                onClickAddScheduleToOrderHandler={sinon.spy()}
                onClickRemoveHandler={sinon.spy()}
            />);

            expect(component).toMatchSnapshot();
        });
    });

    describe('when passed an item', () => {
        test('renders correctly with valid props', () => {
            const component = shallow(<ScheduleItem
                mealScheduleId={mealScheduleItemMock.id}
                isAssembly={false}
                scheduleItem={itemMock}
                onClickAddScheduleToOrderHandler={sinon.spy()}
                onClickRemoveHandler={sinon.spy()}
            />);

            expect(component).toMatchSnapshot();
        });
    });
});
