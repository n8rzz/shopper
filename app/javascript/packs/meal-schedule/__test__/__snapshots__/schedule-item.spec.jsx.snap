// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleItem when passed an assembly renders correctly with valid props 1`] = `
<li
  className="mealSchedule-bd-itemList-item"
>
  <ul
    className="stereo"
  >
    <li>
      Souvlaki
    </li>
    <li>
      <ul
        className="hlist"
      >
        <li
          className="u-isVisuallyHidden"
        >
          <button
            className="icon-btn"
            type="button"
          >
            <span
              className="icon icon-cart-check"
            />
          </button>
        </li>
        <li>
          <button
            className="icon-btn js-mealSchedule-remove"
            onClick={[Function]}
            type="button"
          >
            <span
              className="icon icon-trashcan"
            />
          </button>
        </li>
      </ul>
    </li>
  </ul>
</li>
`;

exports[`ScheduleItem when passed an item renders correctly with valid props 1`] = `
<li
  className="mealSchedule-bd-itemList-item"
>
  <ul
    className="stereo"
  >
    <li>
      Agave Syrup
    </li>
    <li>
      <ul
        className="hlist"
      >
        <li
          className="u-isVisuallyHidden"
        >
          <button
            className="icon-btn"
            type="button"
          >
            <span
              className="icon icon-cart-check"
            />
          </button>
        </li>
        <li>
          <button
            className="icon-btn js-mealSchedule-remove"
            onClick={[Function]}
            type="button"
          >
            <span
              className="icon icon-trashcan"
            />
          </button>
        </li>
      </ul>
    </li>
  </ul>
</li>
`;
